internal/retryer/imdsretryer.go

```diff
@@ -55,3 +56,10 @@ func GetDefaultRetryNumber() int {
 	}
 	return DefaultImdsRetries
 }
+
+// fmtDebugLog logs the content only if the log level is DEBUG
```

Comment:
Thought about dropping this in an util folder but I doubt this is going to be reused considering this is strictly used for the retryer. Any other component used for the agent binary uses OTEL or telegraf loggers.

Reply:
Isn't there a telegraf logger we can use? Instead of adding a function to check the log level?